### PR TITLE
🚀 Add information to app summary output for what will be run on --commit (#27)

### PR DESCRIPTION
**Pull request by Keimya**
**app.py**
In the first invocation without running commit:
- "RAN" is changed to "STAGED"
- "SKIPPED" analyses are separated into SUCCEEDED, FAILED, also indicating if results are protected

In the second invocation upon --commit:
- "RAN" is displayed as the sum of run analyses + unprotected succeeded analyses

scenarios run for protected and unprotected results, and when commit = True and commit = False

**test_app.py**
- def test_commit_description added
- def test_get_experiments_from_default_cli_options modified

- [ ] 🐛 &nbsp; Bug fix
- [x] 🚀 &nbsp; New feature
- [ ] 🔧 &nbsp; Code refactor
- [ ] ⚠️ &nbsp; Breaking change that would cause existing functionality to change
- [ ] 📘 &nbsp; I have updated the documentation accordingly
- [x] ✅ &nbsp; I have added tests to cover my changes

### CODE REVIEW

File: isabl_cli/app.py [hunk 1]
```diff
@@ -893,11 +893,46 @@ def run(
...
+
+            for i in skipped_tuples:
+                if i[1] == "SUCCEEDED":
+                    num_succeeded += 1
+                    if not self.application_protect_results:
+                        num_run_on_commit += 1
+                else:
+                    num_failed += 1
+
+            if num_run_on_commit == 1:
```
Comment (juanesarango) [inline isabl_cli/app.py:922]: should it be `len(num_run_on_commit) == 1`?

**Comment by juanesarango** [comment]
@Keimya do you have examples of how it would look? A screenshot or stdout to add to the PR?

Would `STAGED` been shown twice? We could remove the second print of that one

File: isabl_cli/app.py (hunk 1 above)
Comment (juanesarango) [reply isabl_cli/app.py:922]: Sorry, my bad it's already the `len(run_tuples)`

**Comment by juanesarango** [comment]
Following @funnell's suggestions from #27 what if we do it like this:

**(1) When analyses are protected:**

- First invocation without `--commit`:
```
STAGED 150 | SKIPPED 100 | INVALID 0

Add --commit to run 150 analyses:
```
- Second invocation with `--commit`:

```
RAN 150 | SKIPPED 100 | INVALID 0
```


**(2) When analyses are NOT protected:**

- First invocation without `--commit`:
```
STAGED 150 | SKIPPED 100 | INVALID 0

SKIPPED 100:
    97   SUCCEEDED (Unprotected)
    3     FAILED analyses will be skipped

Add --commit to run 247 analyses:
```
- Second invocation with `--commit`:
```
RAN 247 | SKIPPED 3 | INVALID 0
```

This way in the default scenario where analyses are protected we don't need to detailed each one of the skipped ones saying (Protected).

What do you think? @funnell @Keimya

**Comment by Keimya** [comment]
unprotected results

protected results

**Comment by Keimya** [comment]
updated output:
protected results
protected_results2

unprotected results
unprotected_results2

**Comment by funnell** [comment]
I think I still prefer the message saying something like this for `unprotected`:

```
Add --commit to run 247 analyses:
150  STAGED
97   SUCCEEDED (Unprotected)
```

You could remove the listing of what was skipped if that's too verbose.

**Comment by funnell** [comment]
Another comment I have, that is sort of related, is that I would actually prefer it if Isabl didn't insert analysis rows into the database until `--commit`. The problem I've encountered is misspecifying the filters and all of a sudden there's a bunch of unwanted analysis records.

What do you think of something like:
1. no `--commit`, just check for validity and report how many new analyses would be created, how many unprotected would be run, how many are invalid
2. with `--commit`, create new analysis records, and run them?

**Comment by funnell** [comment]
@juanesarango I just noticed that the cli already prints this at the end of a non-commit run:
```
Add --commit to proceed.
```

so we would end up with something like this?
```
STAGED 1 | SKIPPED 2 | INVALID 0

Add --commit to run 2 analyses:
1  STAGED
1  SUCCEEDED (Unprotected)

Add --commit to proceed.
```

Perhaps the new wording should be changed then? maybe `Add --commit to run 2 analyses:` should be changed to something like `Ready to run 2 analyses:` or `2 analyses available to run:`

**Comment by juanesarango** [comment]
Thanks @Keimya. I like @funnell suggestions. The whole message can be then something like:
```
STAGED 150 | SKIPPED 100 | INVALID 0

247 analyses available to run:
    150  STAGED
    97   SUCCEEDED (Unprotected)

Add --commit to proceed
```

**Comment by Keimya** [comment]
6/16/2023 
Protected
Screenshot 2023-06-16 at 1 34 21 PM

Unprotected
Screenshot 2023-06-16 at 1 34 05 PM

**Review by juanesarango — APPROVED** [review]
Great work @Keimya! Thanks a lot for the contribution! 🙌🏽

**Pull request merged**